src/psbt.c

```diff
@@ -2201,3 +2203,752 @@ int wally_psbt_to_base64(
+        dst->witness_script_len = src->witness_script_len;
+    }
+
+    return WALLY_OK;
+}
+
+int wally_combine_psbts(
+    struct wally_psbt *psbts,
+    size_t psbts_len,
+    struct wally_psbt **output)
+{
+    struct wally_psbt *result;
+    unsigned char global_txid[SHA256_LEN];
+    size_t i, j, k;
+    int ret = WALLY_OK;
+
+    TX_CHECK_OUTPUT;
+
+    // Get info from the first psbt and use it as the template
+    if ((ret = get_txid(psbts->tx, global_txid)) != WALLY_OK) {
+        return ret;
+    }
+
+    if ((ret = wally_psbt_init_alloc(psbts[0].inputs_allocation_len, psbts[0].outputs_allocation_len, psbts[0].unknowns->items_allocation_len, &result)) != WALLY_OK) {
+        return ret;
+    }
+
+    if ((ret = clone_tx(psbts[0].tx, &result->tx)) != WALLY_OK) {
+        goto fail;
+    }
+    result->num_inputs = psbts[0].num_inputs;
+    result->num_outputs = psbts[0].num_outputs;
+
+    for (i = 0; i < psbts_len; ++i) {
+        unsigned char txid[SHA256_LEN];
+
+        // Compare the txids
+        if ((ret = get_txid(psbts[i].tx, txid)) != WALLY_OK) {
+            goto fail;
+        }
+        if (memcmp((char *)global_txid, (char *)txid, SHA256_LEN) != 0) {
+            ret = WALLY_EINVAL;
+            goto fail;
+        }
+
+        // Now start merging
+        for (j = 0; j < result->num_inputs; ++j) {
+            if ((ret = merge_input_into(&result->inputs[j], &psbts[i].inputs[j])) != WALLY_OK) {
+                goto fail;
+            }
+        }
+        for (j = 0; j < result->num_outputs; ++j) {
+            if ((ret = merge_output_into(&result->outputs[j], &psbts[i].outputs[j])) != WALLY_OK) {
+                goto fail;
+            }
+        }
+
+        if (psbts[i].unknowns) {
+            if (!result->unknowns) {
+                if ((ret = wally_unknowns_map_init_alloc(psbts[i].unknowns->items_allocation_len, &result->unknowns)) != WALLY_OK) {
+                    return ret;
+                }
+            }
+
+            for (j = 0; j < psbts[i].unknowns->num_items; ++j) {
+                bool found = false;
+                for (k = 0; k < result->unknowns->num_items; ++k) {
+                    if (result->unknowns->items[k].key_len == psbts[i].unknowns->items[j].key_len &&
+                        memcmp((char *)result->unknowns->items[k].key, (char *)psbts[i].unknowns->items[j].key, psbts[i].unknowns->items[j].key_len) == 0) {
+                        found = true;
+                        break;
+                    }
+                }
+                if (found) {
+                    continue;
+                }
+                if ((ret = add_unknowns_item(result->unknowns, &psbts[i].unknowns->items[j])) != WALLY_OK) {
+                    return ret;
+                }
+            }
+        }
+    }
+
+    *output = result;
+    return WALLY_OK;
+
+fail:
+    if (result) {
+        wally_psbt_free(result);
+    }
+    return ret;
+}
+
+int wally_sign_psbt(
+    struct wally_psbt *psbt,
```

Comment:
should be const

Reply:
The psbt is modified. The signatures are added in place.